front/components/items/FindOrCancelAppointment/index.tsx

```diff
@@ -0,0 +1,199 @@
+                <div className="findorcancelappointment-buttons">
+                    <Button
+                        variant="h3"
+                        colorClass="greendark"
+                        text="Wyszukaj"
+                        onClick={() => {
+                            setIsSearching(true);
+                            setData(undefined);
+                            setFoundAppoinment(undefined);
+                            setIsCanceling(false);
+                            setIsSubmited(false);
+                        }}
+                        className={`findorcancelappointment-buttons__button${
+                            isSearching ? '--active' : ''
+                        }`}
+                    />
+                    <Button
+                        variant="h3"
+                        colorClass="greendark"
+                        text="Odwołaj"
+                        onClick={() => {
+                            setIsSearching(false);
+                            setData(undefined);
+                            setFoundAppoinment(undefined);
+                            setIsCanceling(true);
+                            setIsSubmited(false);
+                        }}
+                        className={`findorcancelappointment-buttons__button${
+                            isCanceling ? '--active' : ''
+                        }`}
+                    />
+                </div>
+                <form onSubmit={formik.handleSubmit} className="findorcancelappointment-form">
+                    <label id="code">Podaj kod wizyty</label>
+                    <div>
+                        <input
+                            id="code"
+                            ref={properCodeInput}
+                            type="text"
+                            value={formik.values.code}
+                            onChange={formik.handleChange}
+                            placeholder="np. 000122"
+                            onInput={validateCode}
+                        />
+                        {formik.errors.code && formik.touched.code && (
+                            <Paragraph text={formik.errors.code} size="small" colorClass="red" />
+                        )}
+                    </div>
+                    <Button
+                        variant="h3"
+                        colorClass="greendark"
+                        text="Sprawdź"
+                        type="submit"
+                        className="findorcancelappointment-form__button"
+                    />
+                </form>
+                {isSearching && (
+                    <div className={!isSubmited ? 'nonedisplay' : undefined}>
+                        <Paragraph
+                            text={
+                                foundAppointment && foundAppointment?.length > 0
+                                    ? foundAppointment
+                                          ?.map((item) => {
```

Comment:
Unnecessary conditional chaining. Youve checked above that foundAppointment is not undefined or null
```suggestion
                                          .map((item) => {
```

Reply:
Also i'd change name 'item' to more meaningful one